domain/src/main/kotlin/no/nav/su/se/bakover/domain/beregning/BeregningMedFradragBeregnetMånedsvis.kt

```diff
@@ -40,41 +41,74 @@ data class BeregningMedFradragBeregnetMånedsvis(
     private fun beregn(): Map<Periode, Månedsberegning> {
         val perioder = periode.tilMånedsperioder()
 
-        val beregnetPeriodisertFradrag = fradragStrategy.beregn(fradrag, periode)
-
-        return perioder.associateWith {
-            MånedsberegningFactory.ny(
-                periode = it,
-                sats = sats,
-                fradrag = beregnetPeriodisertFradrag[it] ?: emptyList(),
-                fribeløpForEps = fradragStrategy.getEpsFribeløp(it)
+        return perioder.associateWith { periode ->
+            beregnMåned(
+                periode = periode,
+                fradrag = fradrag,
             ).let { månedsberegning ->
-                when (månedsberegning.ytelseStørreEnn0MenMindreEnnToProsentAvHøySats()) {
-                    true -> MånedsberegningFactory.ny(
-                        periode = månedsberegning.periode,
-                        sats = sats,
-                        fradrag = månedsberegning.getFradrag()
-                            .plus(månedsberegning.lagFradragForBeløpUnderMinstegrense()),
-                    )
-                    false -> månedsberegning
+                when {
+                    månedsberegning.sosialstønadFørerTilBeløpUnderToProsentAvHøySats(månedsberegning.periode) -> {
+                        månedsberegning.leggTilMerknad(Merknad.Beregning.SosialstønadFørerTilBeløpLavereEnnToProsentAvHøySats)
+                        månedsberegning
+                    }
+                    månedsberegning.beløpStørreEnn0MenMindreEnnToProsentAvHøySats() -> {
+                        beregnMåned(
+                            periode = periode,
+                            fradrag = fradrag + månedsberegning.lagFradragForBeløpUnderMinstegrense(),
+                        ).let {
+                            it.leggTilMerknad(Merknad.Beregning.BeløpMellomNullOgToProsentAvHøySats)
+                            it
+                        }
+                    }
+                    månedsberegning.getSumYtelse() == 0 -> {
+                        månedsberegning.leggTilMerknad(Merknad.Beregning.BeløpErNull)
+                        månedsberegning
+                    }
```

Comment:
Ser ut som här att `Merknad.Beregning.BeløpMellomNullOgToProsentAvHøySats` og `Merknad.Beregning.BeløpErNull` legges kun til når `sosialstønadFørerTilBeløpUnderToProsentAvHøySats` evalueres til true? Kan ha lest feil då alla brackets ligger rart här på github, men tänker att de merknadene burde komma med uavhengig om det er p.g.a sosialstønad eller ikke. 🤪

Reply:
Det stemmer ikke, disse er ikke avhengige av at noe annet har blitt lagt til. De er derimot avhengige av at det _ikke_ er lagt til noe som utelukker seg selv. Dvs; vi identifiserer kun unike tilfeller, enten a eller b eller c. "Og" er ikke et gyldig case for beløp = 0, mellom 0 og 2 og sosialstønad. 

Vi kunne kanskje diskutert navnet på disse, da det kanskje fremstår litt rart at "beregnet = 0" ikke er tilfellet for "beløp mellom 0 og 2 %" hvor sum av ytelsen faktisk er 0. Thoughts?